runtime/autoload/provider/clipboard.vim

```diff
@@ -46,53 +46,117 @@ function! provider#clipboard#Error() abort
   return s:err
 endfunction
 
+let s:providers = [
+      \ {
+      \   'name': 'pbcopy/pbpaste',
+      \   'copy': {
+      \      '+': 'pbcopy',
+      \      '*': 'pbcopy',
+      \    },
+      \   'paste': {
+      \      '+': 'pbpaste',
+      \      '*': 'pbpaste',
+      \   },
+      \   'cache_enabled': 0,
+      \   'check': 'executable("pbcopy")',
+      \ },
+      \ {
+      \   'name': 'xsel',
+      \   'copy': {
+      \      '+': 'xsel --nodetach -i -b',
+      \      '*': 'xsel --nodetach -i -p',
+      \    },
+      \   'paste': {
+      \      '+': 'xsel -o -b',
+      \      '*': 'xsel -o -p',
+      \   },
+      \   'cache_enabled': 1,
+      \   'check': 'exists("$DISPLAY") && executable("xsel") && s:cmd_ok("xsel -o -b")',
+      \ },
+      \ {
+      \   'name': 'xclip',
+      \   'copy': {
+      \      '+': 'xclip -quiet -i -selection clipboard',
+      \      '*': 'xclip -quiet -i -selection primary',
+      \    },
+      \   'paste': {
+      \      '+': 'xclip -o -selection clipboard',
+      \      '*': 'xclip -o -selection primary',
+      \   },
+      \   'cache_enabled': 1,
+      \   'check': 'exists("$DISPLAY") && executable("xclip")',
+      \ },
+      \ {
+      \   'name': 'lemonade',
+      \   'copy': {
+      \      '+': 'lemonade copy',
+      \      '*': 'lemonade copy',
+      \    },
+      \   'paste': {
+      \      '+': 'lemonade paste',
+      \      '*': 'lemonade paste',
+      \   },
+      \   'cache_enabled': 1,
+      \   'check': 'executable("lemonade")',
+      \ },
+      \ {
+      \   'name': 'doitclient',
+      \   'copy': {
+      \      '+': 'doitclient wclip',
+      \      '*': 'doitclient wclip',
+      \    },
+      \   'paste': {
+      \      '+': 'doitclient wclip -r',
+      \      '*': 'doitclient wclip -r',
+      \   },
+      \   'cache_enabled': 1,
+      \   'check': 'executable("doitclient")',
+      \ },
+      \ {
+      \   'name': 'win32yank',
+      \   'copy': {
+      \      '+': 'win32yank -i --crlf',
+      \      '*': 'win32yank -i --crlf',
+      \    },
+      \   'paste': {
+      \      '+': 'win32yank -i --lf',
+      \      '*': 'win32yank -i --lf',
+      \   },
+      \   'cache_enabled': 1,
+      \   'check': 'executable("win32yank")',
+      \ },
+      \ {
+      \   'name': 'tmux',
+      \   'copy': {
+      \      '+': 'tmux load-buffer -',
+      \      '*': 'tmux load-buffer -',
+      \    },
+      \   'paste': {
+      \      '+': 'tmux save-buffer -',
+      \      '*': 'tmux save-buffer -',
+      \   },
+      \   'cache_enabled': 1,
+      \   'check': 'exists("$TMUX") && executable("tmux")',
+      \ },
+      \ ]
+
+function! s:set_provider(p) abort
+    let s:copy = a:p.copy
+    let s:paste = a:p.paste
+    let s:cache_enabled = a:p.cache_enabled
+    return a:p.name
+endfunction
```

Comment:
this isn't actually saving lines since the dictionary definition ends up being 50% bigger. So the churn isn't worth it.